Docs update (#73)
solidity-labs-io/forge-proposal-simulator: docs/overview/use-cases.md

---
description: >-

>     FPS is compatible with the OpenZeppelin Timelock, Compound Governor Bravo, and GnosisSafe Multisig wallet contracts. The project is open source, and we encourage submissions of different types of actions.

Reviewer: ```suggestion
    FPS is compatible with the OpenZeppelin Timelock, Compound Governor Bravo, OZ Governor and GnosisSafe Multisig wallet contracts. The project is open source, and we encourage submissions of different types of actions.
```

---

# Use cases

### OpenZeppelin Timelock Controller

FPS facilitates testing Timelock governance proposals. It supports the creation of both scheduling and execution calldata. Developers can simulate the entire proposal, including scheduling, delays, and execution. Read the [Timelock Proposal](../guides/timelock-proposal.md) documentation to learn how to integrate with the OpenZeppelin Timelock Controller.

### Gnosis Safe Multisig

For Gnosis Safe Multisig support, FPS generates and simulates the Multicall calldata. This allows developers to check protocol health after proposal execution. FPS uses Foundry cheat codes to simulate actions from the actual Multisig address, mirroring real-world proposal execution as much as possible. Calldata generated from this module can be used directly in Gnosis Safe's UI. Check the [Multisig Proposal](../guides/multisig-proposal.md) documentation to learn how to integrate FPS with a Gnosis Safe Multisig.

> ### Compound Governor Bravo

Reviewer: Needs to add OZ Governor Bravo as well

Author: done

# Use cases

> FPS is compatible with the OpenZeppelin Timelock, Compound Governor Bravo, OZ Governor and GnosisSafe Multisig wallet contracts. The project is open source, submission of different types of actions is encouraged.

Reviewer: ```suggestion
The framework is compatible with OpenZeppelin Timelock, Compound Governor Bravo, OZ Governor, and GnosisSafe Multisig wallet contracts. We encourage the submission of pull requests to accommodate different governance models.
```

> The framework is compatible with OpenZeppelin Timelock, Compound Governor Bravo, OZ Governor, and GnosisSafe Multisig wallet contracts. We encourage the submission of pull requests to accommodate different governance models.

Reviewer 2: throughout the docs you change how you refer to OZ Governor by saying Governor OZ and then OZ Governor. I think it should be `OZ Governor` throughout all of the docs and be capitalized as this is a proper noun.

### OpenZeppelin Timelock Controller

> The [Timelock Proposal type](../guides/timelock-proposal.md) facilitates the creation of the scheduling and execution of calldata. It also allows developers to test the calldata by simulating the entire proposal lifecycle, from submission to execution, using foundry cheat codes to bypass the delay period.

Reviewer 2: ```suggestion
The [Timelock Proposal](../guides/timelock-proposal.md) facilitates the creation of the scheduling and execution of calldata. It also allows developers to test the calldata by simulating the entire proposal lifecycle, from submission to execution, using foundry cheat codes to bypass the delay period.
```

> FPS facilitates testing Timelock governance proposals. It supports the creation of both scheduling and execution calldata. Developers can simulate the entire proposal, including scheduling, delays, and execution. Read the [Timelock Proposal](../guides/timelock-proposal.md) documentation to learn how to integrate with the OpenZeppelin Timelock Controller.

Reviewer: ```suggestion
The [Timelock Proposal type](../guides/timelock-proposal.md) facilitates the creation of the scheduling and execution of calldata. It also allows developers to test the calldata by simulating the entire proposal lifecycle, from submission to execution, using foundry cheat codes to bypass the delay period.
```

### Gnosis Safe Multisig

> For Gnosis Safe Multisig support, FPS generates and simulates the Multicall calldata. This allows developers to check protocol health after proposal execution. FPS uses Foundry cheat codes to simulate actions from the actual Multisig address, mirroring real-world proposal execution as much as possible. Calldata generated from this module can be used directly in Gnosis Safe's UI. Check the [Multisig Proposal](../guides/multisig-proposal.md) documentation to learn how to integrate FPS with a Gnosis Safe Multisig.

Reviewer: ```suggestion
The [Multisig Proposal](../guides/multisig-proposal.md) generates and simulates the Multicall calldata. This allows developers to check protocol health after calldata execution by using Foundry cheat codes to simulate actions from the actual Multisig address. Calldata generated from this module can be used directly in Gnosis Safe's UI.
```

### Compound Governor Bravo

> [Governor Bravo Proposal](../guides/governor-bravo-proposal.md) facilitates the creation of the governor `propose` calldata. It also allows developers to test the calldata by simulating the entire proposal lifecycle, from proposing, voting, queuing, and finally executing. 

Reviewer 2: ```suggestion
The [Governor Bravo Proposal](../guides/governor-bravo-proposal.md) facilitates the creation of the governor `propose` calldata. It also allows developers to test the calldata by simulating the entire proposal lifecycle, from proposing, voting, queuing, and finally executing. 
```

### Governor OZ

> Similar to Compound Governor Bravo, FPS simulates the entire proposal lifecycle for the governor with a timelock controller extension. View the [Governor OZ Proposal](../guides/governor-oz-proposal.md) documentation to learn how to integrate with Governor OZ contracts.

Reviewer: ```suggestion
Similar to Compound Governor Bravo, [Governor OZ Proposal](../guides/governor-oz-proposal.md) simulates the entire proposal lifecycle for the governor with a timelock controller extension.
```

> For Compound Governor Bravo, FPS generates and simulates the proposal creation, voting, queuing, and execution. This enables simulation throughout the entire proposal lifecycle for governor and timelock systems. View the [Governor Bravo Proposal](../guides/governor-bravo-proposal.md) documentation to learn how to integrate with Compound Governor Bravo contracts.

Reviewer: ```suggestion
[Governor Bravo Proposal](../guides/governor-bravo-proposal.md) facilitates the creation of the governor `propose` calldata. It also allows developers to test the calldata by simulating the entire proposal lifecycle, from proposing, voting, queuing, and finally executing. 
```
